app.rb

```diff
@@ -0,0 +1,146 @@
+      else
+        puts "\n"
+        peoples.each do |people|
+          puts "ID: #{people.id}"
+          puts "NAME:  #{people.name}"
+          puts "AGE: #{people.age}"
+          puts '========================'
+          puts
+        end
+      end
+    end
+
+    def create_person(person)
+      print "\n Do you want to create a  Student (1) or Teacher (2)? [Input the number]: "
+      option = gets.chomp
+      case option
+      when '1'
+        create_student(person)
+      when '2'
+        create_teacher(person)
+      end
+    end
+
+    def create_student(person)
+      puts "\n"
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Has Parent Permission? [Y/N]: '
+      permission = gets.chomp.upcase
+      print 'Classroom: '
+      classroom = gets.chomp
+
+      parent_permission = permission == 'Y'
+
+      student = Student.new(classroom, age, name, parent_permission: parent_permission)
+      person.push(student)
+      puts "\n Student created successfully!!"
+    end
+
+    def create_teacher(person)
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Specialization: '
+      specialization = gets.chomp
+
+      teacher = Teacher.new(age, specialization, name)
+
+      person.push(teacher)
+
+      puts 'Teacher created successfully!!'
+    end
+  end
+
+  module BookLisitng
+    def list_all_books(books)
+      if books.empty?
+        puts "\n NO BOOKS REGISTERED!"
+      else
+        puts "\n"
+        books.each do |book|
+          puts "Title: #{book.title}"
+          puts "Author #{book.author}"
+          puts '========================'
+          puts
+        end
+      end
+    end
+
+    def create_book(books)
+      print "\nTitle: "
+      title = gets.chomp
+      print 'Author: '
+      author = gets.chomp
+      book = Book.new(title, author)
+      books.push(book)
+      puts "\n Book added sucessfully!"
+    end
+  end
+
+  module RentalListing
+    def list_rentals(rentals, peoples)
+      if rentals.empty?
+        puts "\n NO RENTAL!"
+      else
+        puts "\n Select a person by index"
+        peoples.each_with_index { |p, i| puts "#{i}: #{p.name}" }
+        print 'Person Index: '
+        person_index = gets.chomp.to_i
+
+        name = peoples[person_index].name
+        found = false
+        puts "\n"
+        rentals.each do |rental|
+          next if rental.person.name != name
+
+          found = true
+          puts rental.rentals
+        end
+
+        puts "\nNO RENTAL REGISTERED BY #{name}" unless found
+      end
+    end
+
+    def create_rental(rentals, books, peoples)
+      puts "\n"
+      if @books.empty?
+        puts "\nNO BOOKS TO RENT!\n"
+      elsif @peoples.empty?
+        puts "\nNO PEOPLE TO RENT!\n"
+      else
+        puts 'Select a Book: '
+        @books.each_with_index { |b, i| puts "#{i}: #{b.title} (#{b.author}) " }
+        print 'Book Index: '
```

Comment:
- [ ] The app crashes when I try to create a rental for a book. It shows the error below in the screenshot relating book's author. Please work on that 👍🏽

<img width="1297" alt="Screenshot 2023-04-20 at 10 43 19 PM" src="https://user-images.githubusercontent.com/50941074/233502537-545a2569-b3a1-425e-8dea-041d6b492c36.png">

Reply:
similar issue, with an error in the author, I will fix